src/m365/planner/commands/plan/plan-add.ts

```diff
@@ -63,12 +68,112 @@ class PlannerPlanAddCommand extends GraphCommand {
 
         return request.post(requestOptions);
       })
+      .then(newPlan => this.updatePlanDetails(args.options, newPlan))
       .then((res: any): void => {
         logger.log(res);
         cb();
       }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
   }
 
+  private updatePlanDetails(options: Options, newPlan: PlannerPlan): Promise<PlannerPlan & PlannerPlanDetails> {
+    const planId = newPlan.id as string;
+
+    if (!options.shareWithUserIds && !options.shareWithUserNames) {
+      return Promise.resolve(newPlan);
+    }
+
+    return Promise
+      .all([this.generateSharedWith(options), this.getPlanDetailsEtag(planId)])
+      .then(resArray => {
+        const sharedWith = resArray[0];
+        const etag = resArray[1];
+        const requestOptionsPlanDetails: any = {
+          url: `${this.resource}/v1.0/planner/plans/${planId}/details`,
+          headers: {
+            'accept': 'application/json;odata.metadata=none',
+            'If-Match': etag,
+            'Prefer': 'return=representation'
+          },
+          responseType: 'json',
+          data: {
+            sharedWith: sharedWith
+          }
+        };
+
+        return request.patch(requestOptionsPlanDetails);
+      })
+      .then(planDetails => {
+        return { ...newPlan, ...planDetails as PlannerPlanDetails };
+      });
+  }
+
+  private getPlanDetailsEtag(planId: string): Promise<string> {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/plans/${planId}/details`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get(requestOptions)
+      .then((response: any) => response['@odata.etag']);
+  }
+
+  private generateSharedWith(options: Options): Promise<{ [userId: string]: boolean }> {
+    const sharedWith: { [userId: string]: boolean } = {};
+
+    return this
+      .getUserIds(options)
+      .then((userIds) => {
+        userIds.map(x => sharedWith[x] = true);
+
+        return Promise.resolve(sharedWith);
+      });
+  }
+
+  private getUserIds(options: Options): Promise<string[]> {
+    if (options.shareWithUserIds) {
+      return Promise.resolve(options.shareWithUserIds.split(','));
+    }
+
+    // Hitting this section means assignedToUserNames won't be undefined
+    const userNames = options.shareWithUserNames as string;
+    const userArr: string[] = userNames.split(',').map(o => o.trim());
+    let userIds: string[] = [];
```

Comment:
Since we don't need this variable in this scope, we can move its definition to where we assign it on line 164 and define it as a const instead.